packages/react-instantsearch-hooks/src/lib/useWidget.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useRef } from 'react';
+
+import { dequal } from './dequal';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+import type { Widget } from 'instantsearch.js';
+import type { IndexWidget } from 'instantsearch.js/es/widgets/index/index';
+
+export function useWidget<TWidget extends Widget | IndexWidget, TProps>(
```

Comment:
should we have unit tests for useWidget too, or rely only on integrated tests?

Reply:
It's only extracted because we use the same logic for `useConnector()` and `<Index>`, but it's tested at the public API level.